addon-test-support/@ember/test-helpers/settled.ts

```diff
@@ -158,8 +162,8 @@ export function getSettledState() {
   let pendingRequestCount = pendingRequests();
 
   return {
-    hasPendingTimers: Boolean(run.hasScheduledTimers()),
-    hasRunLoop: Boolean(run.currentRunLoop),
+    hasPendingTimers: Boolean((run as any).hasScheduledTimers()),
```

Comment:
Doesn't `Ember.run` have stuff in `@types/ember`? Why do we have to cast here?

Reply:
it does, but unfortunately those two are missing from the published types 😢

Reply:
@Turbo87 @rwjblue The only reason they're missing is because [there's no indication that they're part of Ember's public API](https://github.com/emberjs/ember.js/blob/d277c93bc0d357ff50e2c62ec1853794a9b82dd1/packages/%40ember/runloop/index.js#L337-L339). We look to the https://www.emberjs.com/api/ as the source of truth.

Reply:
Ya, seems fine though this library (IMHO by necessity) will have to use what amount to private APIs for applications so that it can absorb any private API churn and expose _actually_ public stable APIs to our users.

Reply:
> Ya, seems fine though this library (IMHO by necessity) will have to use what amount to private APIs for applications so that it can absorb any private API churn and expose _actually_ public stable APIs to our users.

In the future, we may want to discuss the pros/cons of publishing an additional types package for authors of libraries like this, to layer on top of the public ember types. I have some quality/stability concerns around each author aiming to define/maintain this type info themselves.

Reply:
Yeah. The issue is somewhat complex, the layers of “public” are quite different. For example:

* intended for consumption by apps
* intended for consumption by tooling
* intended for consumption by “privileged” addons

Things like the modifier and component manager APIs are good examples of things intended for addons but not apps. 

Things like default outlet templates / using outlet state / etc are examples of things Ember intentionally exposes for this addon to be able to provide its APIs, but those things should never be made exposed to other addons or apps...